src/OmniSharp.MSBuild/MSBuildProjectSystem.cs

```diff
@@ -330,7 +350,13 @@ private void UpdateSourceFiles(Project project, IList<string> sourceFiles)
                     continue;
                 }
 
-                // If not, add a new document.
+                // If the source file doesn't exist on disk, don't try to add it.
+                if (!File.Exists(sourceFile))
```

Comment:
This is avoiding a crash below. Unless atom/vscode show you a view of your project, I'm not sure it matters whether we do something fancy in the workspace. This is the scenario where VS would show you a little warning icon in the solution explorer.